docs/docs/30-administration/11-forges/60-bitbucket_datacenter.md

```diff
@@ -0,0 +1,171 @@
+
+Private key file mounted into your Woodpecker container at runtime as a volume.
+
+```diff
+# docker-compose.yml
+version: '3'
+
+services:
+  woodpecker-server:
+    [...]
+    environment:
+      - [...]
+      - WOODPECKER_BITBUCKET_DC=true
+      - WOODPECKER_BITBUCKET_DC_GIT_USERNAME=foo
+      - WOODPECKER_BITBUCKET_DC_GIT_PASSWORD=bar
+      - WOODPECKER_BITBUCKET_DC_CONSUMER_KEY=95c0282573633eb25e82
++     - WOODPECKER_BITBUCKET_DC_CONSUMER_RSA=/etc/bitbucket/key.pem
+      - WOODPECKER_BITBUCKET_DC_URL=http://stash.mycompany.com
++  volumes:
++     - /etc/bitbucket/key.pem:/etc/bitbucket/key.pem
+
+  woodpecker-agent:
+    [...]
+```
+
+Private key as environment variable
+
+```diff
+# docker-compose.yml
+version: '3'
+
+services:
+  woodpecker-server:
+    [...]
+    environment:
+      - [...]
+      - WOODPECKER_BITBUCKET_DC=true
+      - WOODPECKER_BITBUCKET_DC_GIT_USERNAME=foo
+      - WOODPECKER_BITBUCKET_DC_GIT_PASSWORD=bar
+      - WOODPECKER_BITBUCKET_DC_CONSUMER_KEY=95c0282573633eb25e82
++     - WOODPECKER_BITBUCKET_DC_CONSUMER_RSA_STRING=contentOfPemKeyAsString
+      - WOODPECKER_BITBUCKET_DC_URL=http://stash.mycompany.com
+
+  woodpecker-agent:
+    [...]
+```
+
+## Service Account
+
+Woodpecker uses `git+https` to clone repositories, however, Bitbucket Server does not currently support cloning repositories with OAuth token. To work around this limitation, you must create a service account and provide the username and password to Woodpecker. This service account will be used to authenticate and clone private repositories.
+
+## Registration
+
+You must register your application with Bitbucket Server in order to generate a consumer key. Navigate to your account settings and choose Applications from the menu, and click Register new application. Now copy & paste the text value from `/etc/bitbucket/key.pub` into the `Public Key` in the incoming link part of the application registration.
+
+Please use the following as the Authorization callback URL:
+
+```nohighlight
+http://<your-woodpecker-address>/authorize
+```
+
+## Configuration
+
+This is a full list of configuration options. Please note that many of these options use default configuration values that should work for the majority of installations.
+
+### `WOODPECKER_BITBUCKET_DC`
+
+> Default: `false`
+
+Enables the Bitbucket Server driver.
+
+### `WOODPECKER_BITBUCKET_DC_URL`
+
+> Default: empty
+
+Configures the Bitbucket Server address.
+
+### `WOODPECKER_BITBUCKET_DC_CONSUMER_KEY`
+
+> Default: empty
+
+Configures your Bitbucket Server consumer key.
+
+### `WOODPECKER_BITBUCKET_DC_CONSUMER_KEY_FILE`
+
+> Default: empty
+
+Read the value for `WOODPECKER_BITBUCKET_DC_CONSUMER_KEY` from the specified filepath
+
+### `WOODPECKER_BITBUCKET_DC_CONSUMER_RSA`
+
+> Default: empty
+
+Configures the path to your Bitbucket Server private key file.
+
+### `WOODPECKER_BITBUCKET_DC_CONSUMER_RSA_STRING`
```

Comment:
Now by looking at it I would swap those two to have it consistent to the other options:

`WOODPECKER_BITBUCKET_DC_CONSUMER_RSA` for the string and `WOODPECKER_BITBUCKET_DC_CONSUMER_RSA_FILE` to read from a file